staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -143,9 +146,30 @@ func (s *ServerRunOptions) Validate() []error {
 		errors = append(errors, fmt.Errorf("--max-resource-write-bytes can not be negative value"))
 	}
 
+	if err := validateHSTSHeader(s.HSTSHeaders); err != nil {
+		errors = append(errors, err)
+	}
 	return errors
 }
 
+func validateHSTSHeader(hstsHeaders []string) error {
+	// HSTS Headers format: Strict-Transport-Security:max-age=expireTime [;includeSubDomains] [;preload]
+	if len(hstsHeaders) != 0 {
+		for _, hstsHeader := range hstsHeaders {
+			if len(strings.TrimSpace(hstsHeader)) == 0 {
+				return fmt.Errorf("empty value in strict-transport-security-headers")
```

Comment:
build a full slice of errors `[]error` and then use `pkg/util/errors.NewAggregate` at the end.  If memory serves, it correctly results nil on an empty slice, so you get the "easy" behavior that works and all the errors at the same time.